jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ExtUtils.java

```diff
@@ -35,10 +65,186 @@ public static String toUri(JarEntryFile jarEntryFile) {
         }
     }
 
+    public static boolean isJarResourceUri(URI uri) {
+        return uri != null && JDT_SCHEME.equals(uri.getScheme()) && CONTENTS_AUTHORITY.equals(uri.getAuthority());
+    }
+
+    public static IJarEntryResource createJarResource(URI uri) throws JavaModelException {
```

Comment:
This function name is misleading. You didn't create any resource here. The function just enumerate all resource files and return the list.